File: src/client/flutter/lib/models/entities/response_api.dart
class ResponseAPI {
  String status;
  String? message;
  int code;
  dynamic data;

  ResponseAPI({
    required this.status,
    this.message,
    required this.code,
    this.data,
  });

  ResponseAPI.fromJson(Map<String, dynamic> json)
      : status = json["status"],
        message = json["message"],
        code = json["code"],
        data = json["data"];

}

File: src/app/android/app/src/main/java/com/cyberacy/app/models/entities/ResponseAPI.kt
package com.cyberacy.app.models.entities

import com.google.gson.annotations.SerializedName

class ResponseAPI<T>(
    @SerializedName("status")
    val status: String,

    @SerializedName("message")
    val message: String?,

    @SerializedName("code")
    val code: Int,

    @SerializedName("data")
    val data: T?,
) { }